share/shwap/p2p/bitswap/block_store.go

```diff
@@ -38,6 +38,11 @@ func (b *Blockstore) getBlock(ctx context.Context, cid cid.Cid) (blocks.Block, e
 	if err != nil {
 		return nil, fmt.Errorf("getting EDS Accessor for height %v: %w", blk.Height(), err)
 	}
+	defer func() {
+		if err := eds.Close(); err != nil {
+			log.Warnf("failed to close EDS accessor for height %v: %s", blk.Height(), err)
+		}
+	}()
```

Comment:
We now have handy utils.CloseAndLog for that, as it becomes common code accross the repo

Reply:
Its not merged to shwap yet and i dont find it useful honestly. I am gonna keep it as it is here

Reply:
The log message should contain the context of where the warn/error happens. Using this util will remove any context

Reply:
It takes logger as first argument, so you can pass as much additional context information as you want within loggier fields.

Reply:
This is optional, so  would defer to you to decide.

Reply:
i dont see any value so gonna keep the way it is